File: src/main/java/com/eager/questioncloud/verification/repository/EmailVerificationJpaRepository.java
package com.eager.questioncloud.verification.repository;

import com.eager.questioncloud.verification.entity.EmailVerificationEntity;
import com.eager.questioncloud.verification.vo.EmailVerificationType;
import jakarta.persistence.Tuple;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

public interface EmailVerificationJpaRepository extends JpaRepository<EmailVerificationEntity, String> {
    Optional<EmailVerificationEntity> findByTokenAndEmailVerificationTypeAndIsVerifiedFalse(
        String token, EmailVerificationType emailVerificationType);

    @Query(value = "SELECT e as emailVerification, u as user from EmailVerificationEntity e left join UserEntity u on u.uid = e.uid where e.resendToken =:resendToken and e.isVerified = false")
    Optional<Tuple> findByResendTokenWithUser(String resendToken);

    Optional<EmailVerificationEntity> findByUidAndIsVerifiedFalse(Long uid);
}

File: qc-core/qc-user/qc-user-core/src/main/kotlin/com/eager/questioncloud/user/infrastructure/repository/EmailVerificationJpaRepository.kt
package com.eager.questioncloud.user.infrastructure.repository

import com.eager.questioncloud.user.enums.EmailVerificationType
import com.eager.questioncloud.user.infrastructure.entity.EmailVerificationEntity
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface EmailVerificationJpaRepository : JpaRepository<EmailVerificationEntity, String> {
    fun findByTokenAndEmailVerificationTypeAndIsVerifiedFalse(
        token: String,
        emailVerificationType: EmailVerificationType
    ): Optional<EmailVerificationEntity>

    fun findByResendToken(resendToken: String): Optional<EmailVerificationEntity>
}
